mcp2210-core.c

```diff
@@ -709,6 +712,358 @@ static inline int eeprom_read_complete(struct mcp2210_cmd *cmd_head,
+                        &value))
+                board_config._3wire_tx_enable_active_high = value;
+
+            of_property_read_u8(temp, "_3wire_tx_enable_pin",
+                    &board_config._3wire_tx_enable_pin);
+            of_property_read_u32(temp, "strings_size",
+                    &board_config.strings_size);
+
+            /* Parse all the pin configurations */
+            for_each_child_of_node (temp, np_pin)
+            {
+                struct device_node *spi_node;
+                u8 pin;
+                if (of_property_read_u8(np_pin, "pin", &pin))
+                {
+                    mcp2210_err("pin property not found, ignoring rest of this child\n");
+                    continue;
+                }
+                if (pin > MCP2210_NUM_PINS) {
+                    mcp2210_err("Invalid pin number %d in pin node\n", pin);
+                    continue;
+                }
+                /* Bit field pointer can not be passed directly to
+                 * of_property_* calls */
+                if (0 == of_property_read_u8(np_pin, "mode", &value))
+                    board_config.pins[pin].mode = value;
+                if (0 == of_property_read_u8(np_pin, "has_irq", &value))
+                    board_config.pins[pin].has_irq = value;
+                if (0 == of_property_read_u8(np_pin, "irq", &value))
+                    board_config.pins[pin].irq = value;
+                if (0 == of_property_read_u8(np_pin, "irq_type", &value))
+                    board_config.pins[pin].irq_type = value;
+
+                if (of_property_read_string(np_pin, "pin_name",
+                            &board_config.pins[pin].name))
+                    board_config.pins[pin].name = NULL;
+                if (of_property_read_string(np_pin, "modalias",
+                            &board_config.pins[pin].modalias))
+                    board_config.pins[pin].modalias = NULL;
+
+                /* Parse spi settings, if available for this pin */
+                spi_node = of_get_child_by_name(np_pin, "spi_prop");
+                if (spi_node)
+                {
+                    of_property_read_u32(spi_node, "spi,max_speed_hz",
+                            &board_config.pins[pin].spi.max_speed_hz);
+                    of_property_read_u32(spi_node, "spi,min_speed_hz",
+                            &board_config.pins[pin].spi.min_speed_hz);
+                    of_property_read_u8(spi_node, "spi,mode",
+                            &board_config.pins[pin].spi.mode);
+                    of_property_read_u8(spi_node, "spi,bits_per_word",
+                            &board_config.pins[pin].spi.bits_per_word);
+
+                    /* Bit field pointer can not be passed directly to
+                     * of_property_* calls */
+                    if (0 == of_property_read_u8(spi_node, "spi,use_cs_gpio",
+                                &value))
+                        board_config.pins[pin].spi.use_cs_gpio = value;
```

Comment:
I don't remember the reason that's a bitfield, except that it may have originally shared it's byte with some other flags. We can probably just changed it to a u8 and modify any place it's read if it expects a 1 or zero.